src/components/products/ProductNav.jsx

```diff
@@ -1,18 +1,403 @@
+          brandmorelessText: "Less"
+        });
+      }
+      else if (this.state.isBrandMoreClicked === true) {
+        this.setState({
+          isBrandMoreClicked: !this.state.isBrandMoreClicked,
+          brandmorelessText: "More"
+        });
+      }
+    }
+
+    else if (clickValue === "icon") {
+      this.setState({
+        iconVisibility: !this.state.iconVisibility
+      });
+    }
+
+    else if (clickValue === "iconmoreless") {
+      if (this.state.isIconMoreClicked === false) {
+        this.setState({
+          isIconMoreClicked: !this.state.isIconMoreClicked,
+          iconmorelessText: "Less"
+        });
+      }
+      else if (this.state.isIconMoreClicked === true) {
+        this.setState({
+          isIconMoreClicked: !this.state.isIconMoreClicked,
+          iconmorelessText: "More"
+        });
+      }
+    }
+
+  }
+
   render() {
+
+    const genderIcon = this.state.genderVisibility === true ? CaretUpIcon : CaretDownIcon;
+    const kidsIcon = this.state.kidsVisibility === true ? CaretUpIcon : CaretDownIcon;
+    const collabIcon = this.state.collabVisibility === true ? CaretUpIcon : CaretDownIcon;
+    const bestforIcon = this.state.bestforVisibility === true ? CaretUpIcon : CaretDownIcon;
+    const priceIcon = this.state.priceVisibility === true ? CaretUpIcon : CaretDownIcon;
+    const sportsIcon = this.state.sportsVisibility === true ? CaretUpIcon : CaretDownIcon;
+    const athleteIcon = this.state.athleteVisibility === true ? CaretUpIcon : CaretDownIcon;
+    const brandIcon = this.state.brandVisibility === true ? CaretUpIcon : CaretDownIcon;
+    const iconIcon = this.state.iconVisibility === true ? CaretUpIcon : CaretDownIcon;
+
+    const morelessIcon = this.state.isMoreClicked === true ? MinusIcon : PlusIcon;
+    const sportmorelessIcon = this.state.isSportMoreClicked === true ? MinusIcon : PlusIcon;
+    const athletemorelessIcon = this.state.isAthleteMoreClicked === true ? MinusIcon : PlusIcon;
+    const brandmorelessIcon = this.state.isBrandMoreClicked === true ? MinusIcon : PlusIcon;
+    const iconmorelessIcon = this.state.isIconMoreClicked === true ? MinusIcon : PlusIcon;
+
     return (
-      <Pane className="product-nav">
-        <Pane >
-          <h2>Product search & sort here</h2>
-        </Pane>
-        <Pane aria-label="Radio Group Label 12" role="group">
-          <Radio name="group" label="Dress" />
-          <Radio name="group" checked label="Pants" />
-          <Radio name="group" disabled label="Shoes" />
-          <Radio checked name="group" label="Undergarments" />
+      <Pane className="product-filter-modal">
+        <Pane className="product-subcateogry-links">
+          <UnorderedList className="subcategory-list-items">
+            {this.state.subCategoryLinks.map(subCategory => <ListItem className="filter-list-items" key={subCategory.id}><a className="filter-list-tags" href={subCategory.link}>{subCategory.title}</a></ListItem>)}
```

Comment:
Instead of href here use link/navlink as we are using react-routning to redirect to a new page. href will refresh the page  while redirecting but link/navlink will make it feel like a multipage website in a Single page as it will not refresh